R/html_document.R

```diff
@@ -81,28 +83,50 @@
 #'@return R Markdown output format to pass to \code{\link{render}}
 #'
 #'@section Anchor Sections Customization:
-#'  By default, a \samp{#} is used as a minimalist choice, referring to the id selector
-#'  in HTML and CSS. You can easily change that using a css rule in your
-#'  document. For example, to add a \href{https://codepoints.net/U+1F517}{link
-#'  symbol} \if{html}{\out{(&#x1F517;&#xFE0E;)}} instead:
+#'  This will be the default to activate anchor sections link on header
+#'  \preformatted{
+#'  output:
+#'    html_document:
+#'      anchor_sections: TRUE
+#'  }
+#'  There is currently two options to modify the default behavior
+#'
+#'  \describe{
+#'  \item{\code{style}}{It allows to select a predefined style:
+#'  \itemize{
+#'  \item{\code{style = "dash"}, the default, is using a \samp{#}, a minimalist choice referring to the id selector in HTML and CSS.}
+#'  \item{\code{style = "symbol"} will use a \href{https://codepoints.net/U+1F517}{link
+#'  symbol} \if{html}{\out{(&#x1F517;&#xFE0E;)}}}
+#'  \item{\code{style = "icon"} will use an svg icon from \url{https://material.io/resources/icons/}
+#'  \if{html}{(\figure{link-black-18dp.svg}{options: alt="icon link"})}}
+#'  }
+#'  You can also easily change to any of your choosing using a css rule in your
+#'  document. For example, to get a pictogram \if{html}{\out{(&#x1F517;)}}:
```

Comment:
Isn't this the link symbol?

Reply:
It is but use a pictogram that could differ according to the browser. We use the symbol version instead by using the special variation selector 
> Emoji-specific variation selector: text, signals intention for preceding glyph to be rendered with basic line stroke rather than complex pictogram.
Source: 
* https://graphemica.com/%EF%B8%8E
* https://codepoints.net/U+1F517

Example: 
![image](https://user-images.githubusercontent.com/6791940/102621577-0dca3480-4140-11eb-8bee-30da9568230d.png)

The documentation was a way to show how to use the pictogram if one desires

Reply:
Ah, I didn't spot that you were using a specific technical unicode term. Given the additional complexity of explaining all these details, I'd be tempted to pick a different example. (Especially since I think most people would want 🔗).

Reply:
> Especially since I think most people would want 🔗

As it can be rendered differently depending on OS (Ex. of differences https://www.utf8icons.com/character/128279/link-symbol), I wanted to use a default for `symbol` with less variations. If someone wants the unicode icon it can be added using CSS as in example. But for icon, I thought the svg one was a better fit.

Now I am not sure which default is better and I wonder if offering the choice between `symbol` and `icon` is really interesting - they are similar.

Reply:
I realised that the other downside of the emoji is that it won't get coloured by css, so I think the best solution really is the svg.